src/nodelist.rs

```diff
@@ -48,7 +48,7 @@ struct NodelistParser<'a> {
     i: usize,
 }
 
-impl<'a> NodelistParser<'a> {
+impl NodelistParser<'_> {
```

Comment:
TIL!  I ran into this clippy remark myself and was stumped at how to fix it.  I've no idea why this fix makes sense, but I'm OK with it.